### PR TITLE
Draft ExtensionArray post

### PR DESCRIPTION
**Pull request by TomAugspurger**

### CODE REVIEW

File: _posts/2019-01-16-dask-extension-arrays.md [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+dtype: datetime64[ns, US/Central]
+Dask Name: from_pandas, 2 tasks
+```
+
+The Challenge
+-------------
+
+Newer versions of pandas allow third-party libraries to write custom extension
+arrays. These arrays can be placed inside a DataFrame or Series, and work
+just as well as any extension arary defined within pandas itself. However,
```
Comment (mrocklin) [inline _posts/2019-01-16-dask-extension-arrays.md]:
```suggestion
just as well as any extension array defined within pandas itself. However,
```

File: _posts/2019-01-16-dask-extension-arrays.md [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+
+Dask DataFrame works well with pandas' new Extension Array interface, including
+third-party extension arrays.
+
+Background
+----------
+
+Pandas 0.23 introduced the [`ExtensionArray`][EA], a way to store things other
+than a simple NumPy array in a DataFrame or Series. Internally pandas uses this
+for data types that aren't handled natively by NumPy (datetimes with timezones,
```
Comment (mrocklin) [inline _posts/2019-01-16-dask-extension-arrays.md]:
```suggestion
for data types that aren't handled natively by NumPy like datetimes with timezones,
```

File: _posts/2019-01-16-dask-extension-arrays.md [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+Categorical, and nullable integer arrays, to name a few).
```
Comment (mrocklin) [inline _posts/2019-01-16-dask-extension-arrays.md]:
```suggestion
Categorical, and nullable integer arrays (new!).
```

File: _posts/2019-01-16-dask-extension-arrays.md [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+with the community. If you visit [dask.org](https://dask.org), you'll see
+phrases like
+
+> Integrates with existing projects
+
+and
+
+> Built with the broader community
+
+At the start of Dask, the developers *could* have gone off an re-written pandas
```
Comment (mrocklin) [inline _posts/2019-01-16-dask-extension-arrays.md]:
```suggestion
At the start of Dask, the developers *could* have gone off and re-written pandas
```

File: _posts/2019-01-16-dask-extension-arrays.md [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+
+And now, when pandas introduces new features like nullable integers,
+`dask.dataframe` just needs to register it as an extension type and immediately
+benefits from it. And third-party extension array authors can do the same for
+their extension arrays.
+
+If you're writing an ExtensionArray, make sure to add it to the [pandas
+ecosystem](http://pandas.pydata.org/pandas-docs/version/0.24/ecosystem.html#extension-data-types)
+page, and register it with Dask.
+
```
Comment (mrocklin) [inline _posts/2019-01-16-dask-extension-arrays.md]:
```suggestion

If you're looking to contribute on a fun project, we recommend trying to connect Dask dataframe up to an existing extension array like [cyberpandas](https://github.com/ContinuumIO/cyberpandas) or TODO.  

```

Comment (mrocklin) [reply _posts/2019-01-16-dask-extension-arrays.md]: Also, should we mention these sorts of projects at the beginning to add to the motivation?  Currently all of the new functionality we present is decimal.

**Review by mrocklin — COMMENTED** [review]
This looks great.  I enjoyed the read.  I added a few tiny suggestions and a request to highlight some practical applications like cyberpandas earlier on.

**Comment by TomAugspurger** [comment]
Thank, updated.

I reworked the "The Real Lesson" section a bit more to emphasize

1. Dask benefits directly from improvements to pandas (like the new nullable integer type)
2. 3rd party EAs (like cyberpandas) don't have to reimplement things from scratch, just to work with dask.

**Comment by mrocklin** [comment]
@TomAugspurger OK to update the date in the filename and then merge this?

**Comment by TomAugspurger** [comment]
Sure. Done.

**Pull request merged**